src/constbuffer_array.c

```diff
@@ -433,7 +433,7 @@ IMPLEMENT_MOCKABLE_FUNCTION(, void, constbuffer_array_dec_ref, CONSTBUFFER_ARRAY
     else
     {
         /* Codes_SRS_CONSTBUFFER_ARRAY_01_016: [ Otherwise constbuffer_array_dec_ref shall decrement the reference count for constbuffer_array_handle. ]*/
-        if (DEC_REF(CONSTBUFFER_ARRAY_HANDLE_DATA, constbuffer_array_handle) == DEC_RETURN_ZERO)
+        if (DEC_REF(CONSTBUFFER_ARRAY_HANDLE_DATA, constbuffer_array_handle) == 0)
```

Comment:
>0 [](start = 80, length = 1)

Seems like if we're still using DEC_REF here, we should keep using DEC_RETURN_ZERO, right? IIRC that was defined in REF_COUNT so that ref count owns which is the actual value when the reference count is 0. Just looks a bit odd to change it only here in one place.

Reply:
Your memories are accurate, however, recently refcount_os went away. And DEC_REF is now interlocked_xxx based. And interlocked_decrement returns 0. And DEC_RETURN_ZERO doesn't even exist anymore. (we had DEC_RETURN_ZERO for OS reasons, C11 whatever).

But Parth's work enabled us to stop caring about OS in interlocked_xxx so why not use interlocked_xxx for refcounting? :) [and I need it on a separate track with gballoc that will be revealed in a future CR]

---
In reply to: [460211383](https://github.com/Azure/azure-c-util/pull/51#discussion_r460211383) [](ancestors = 460211383)